src/main/java/duke/task/Task.java

```diff
@@ -0,0 +1,156 @@
+package duke.task;
+
+import duke.exception.DukeException;
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
+/**
+ * Represents a task. A <code>Task</code> consist of
+ * name, type (todo, event, deadline), status and deadline (if applicable).
+ */
+public class Task {
+    private String name;
+    private boolean done;
+    private String type;
+    private String preposition;
+    private LocalDate date;
+
+    /**
+     * Task constructor for todo.
+     *
+     * @param type
+     * @param name
+     */
+    public Task(String type, String name) {
+        this.name = name;
+        this.done = false;
+        this.type = type;
+        this.date = null;
+    }
+
+    /**
+     * Task constructor for event/deadline.
+     *
+     * @param type
+     * @param name
+     * @param date
+     * @param preposition
+     */
+    public Task(String type, String name, String date, String preposition) {
+        this.name = name;
+        this.done = false;
+        this.type = type;
+        this.date = LocalDate.parse(date);
+        this.preposition = preposition;
+    }
+
+    /**
+     * Getter method for type.
+     *
+     * @return the type of the <code>Task</code>
+     */
+    public String getType() {
+        return type;
+    }
+
+    /**
+     * Getter method for status.
+     *
+     * @return the status of the <code>Task</code>
+     */
+    public boolean getDone() {
+        return done;
+    }
+
+    /**
+     * Getter method for date.
+     *
+     * @return the date of the <code>Task</code>
+     */
+    public LocalDate getDate() {
+        return date;
+    }
+
+    /**
+     * Getter method for name.
+     *
+     * @return the name of the <code>Task</code>
+     */
+    public String getName() {
+        return name;
+    }
+
+    /**
+     * Setter method for status.
+     * This method set the <code>Task</code> status to done.
+     */
+    public void setDone() {
+        this.done = true;    
+    }
+
+    /**
+     * Returns a readable string representing the task that can
+     * be used for outputting to user.
+     *
+     * @return a representative string of <code>Task</code>
+     */
+    @Override
+    public String toString() {
+        if (type.equals("E") || type.equals("D")) {
+            return String.format("[%s][%s] %s (%s: %s)",
+                    type,
+                    done ? "X" : " ",
```

Comment:
Perhaps make the code more obvious here by adding parentheses around the same grouping?